app/src/main/java/com/termux/filepicker/TermuxFileReceiverActivity.java

```diff
@@ -0,0 +1,202 @@
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.regex.Pattern;
+
+public class TermuxFileReceiverActivity extends Activity {
+
+    static final String TERMUX_RECEIVEDIR = TermuxConstants.TERMUX_FILES_DIR_PATH + "/home/downloads";
+    static final String EDITOR_PROGRAM = TermuxConstants.TERMUX_HOME_DIR_PATH + "/bin/termux-file-editor";
+    static final String URL_OPENER_PROGRAM = TermuxConstants.TERMUX_HOME_DIR_PATH + "/bin/termux-url-opener";
+
+    /**
+     * If the activity should be finished when the name input dialog is dismissed. This is disabled
+     * before showing an error dialog, since the act of showing the error dialog will cause the
+     * name input dialog to be implicitly dismissed, and we do not want to finish the activity directly
+     * when showing the error dialog.
+     */
+    boolean mFinishOnDismissNameDialog = true;
+
+    private static final String LOG_TAG = "TermuxFileReceiverActivity";
+
+    static boolean isSharedTextAnUrl(String sharedText) {
+        return Patterns.WEB_URL.matcher(sharedText).matches()
+            || Pattern.matches("magnet:\\?xt=urn:btih:.*?", sharedText);
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+
+        final Intent intent = getIntent();
+        final String action = intent.getAction();
+        final String type = intent.getType();
+        final String scheme = intent.getScheme();
+
+        if (Intent.ACTION_SEND.equals(action) && type != null) {
+            final String sharedText = intent.getStringExtra(Intent.EXTRA_TEXT);
+            final Uri sharedUri = intent.getParcelableExtra(Intent.EXTRA_STREAM);
+
+            if (sharedText != null) {
+                if (isSharedTextAnUrl(sharedText)) {
+                    handleUrlAndFinish(sharedText);
+                } else {
+                    String subject = intent.getStringExtra(Intent.EXTRA_SUBJECT);
+                    if (subject == null) subject = intent.getStringExtra(Intent.EXTRA_TITLE);
+                    if (subject != null) subject += ".txt";
+                    promptNameAndSave(new ByteArrayInputStream(sharedText.getBytes(StandardCharsets.UTF_8)), subject);
+                }
+            } else if (sharedUri != null) {
+                handleContentUri(sharedUri, intent.getStringExtra(Intent.EXTRA_TITLE));
+            } else {
+                showErrorDialogAndQuit("Send action without content - nothing to save.");
+            }
+        } else if ("content".equals(scheme)) {
+            handleContentUri(intent.getData(), intent.getStringExtra(Intent.EXTRA_TITLE));
+        } else if ("file".equals(scheme)) {
+            // When e.g. clicking on a downloaded apk:
+            String path = intent.getData().getPath();
+            File file = new File(path);
+            try {
+                FileInputStream in = new FileInputStream(file);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *in*. Execution paths that do not contain closure statements were detected. Ensure *in* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).